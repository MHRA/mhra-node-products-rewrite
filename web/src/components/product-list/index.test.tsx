import renderer from 'react-test-renderer';
import React from 'react';
import ProductList from './index';

describe(ProductList, () => {
  it('should render', () => {
    const component = renderer
      .create(
        <ProductList
          title={'Coffee'}
          products={[
            { name: 'Caffe Latte', count: 0 },
            { name: 'Americano', count: 1 },
            { name: 'Flat White', count: -1 },
            { name: 'Mocha', count: 9999999 },
            { name: 'Cafe Creme', count: -9999999 },
          ]}
        />,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
  it('should render error message', () => {
    const component = renderer
      .create(
        <ProductList
          title={'Coffee'}
          products={[
            { name: 'Caffe Latte', count: 0 },
            { name: 'Americano', count: 1 },
            { name: 'Flat White', count: -1 },
            { name: 'Mocha', count: 9999999 },
            { name: 'Cafe Creme', count: -9999999 },
          ]}
          errorFetchingResults
        />,
      )
      .toJSON();
    expect(component).toMatchSnapshot();
  });
  it('should render loading message', () => {
    const component = renderer
      .create(<ProductList title={'Coffee'} products={[]} isLoading />)
      .toJSON();
    expect(component).toMatchSnapshot();
  });
});
