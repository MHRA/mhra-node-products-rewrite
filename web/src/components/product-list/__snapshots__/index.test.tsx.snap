// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductList should render 1`] = `
<nav
  className="sc-bczRLJ cVndyS"
>
  <h2>
    Coffee
  </h2>
  <ul>
    <li
      className="product-name"
    >
      <a
        href="/product?product=Caffe%20Latte"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Caffe Latte
         
        0
      </a>
    </li>
    <li
      className="product-name"
    >
      <a
        href="/product?product=Americano"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Americano
         
        (
        1
         
        file
        )
      </a>
    </li>
    <li
      className="product-name"
    >
      <a
        href="/product?product=Flat%20White"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Flat White
         
        (
        -1
         
        files
        )
      </a>
    </li>
    <li
      className="product-name"
    >
      <a
        href="/product?product=Mocha"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Mocha
         
        (
        9999999
         
        files
        )
      </a>
    </li>
    <li
      className="product-name"
    >
      <a
        href="/product?product=Cafe%20Creme"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        Cafe Creme
         
        (
        -9999999
         
        files
        )
      </a>
    </li>
  </ul>
</nav>
`;

exports[`ProductList should render error message 1`] = `
<nav
  className="sc-bczRLJ cVndyS"
>
  <p
    className="sc-gsnTZi dPidEJ"
  >
    Sorry - the site is experiencing technical issues right now. Please try again later.
  </p>
</nav>
`;

exports[`ProductList should render loading message 1`] = `
<nav
  className="sc-bczRLJ cVndyS"
>
  <h2>
    Coffee
  </h2>
  <ul>
    <li>
      Loading results...
    </li>
  </ul>
</nav>
`;
